frontend/src/hoc/withFeatureFlag.tsx

```diff
@@ -20,16 +20,17 @@ const withFeatureFlag = <P, R>(
   }
 
   // top level component to grab search params from the top level page props
-  const ComponentWithFeatureFlagAndSearchParams = (
+  const ComponentWithFeatureFlagAndSearchParams = async (
     props: P & WithFeatureFlagProps,
   ) => {
-    const searchParams = props.searchParams || {};
+    const searchParams = (await props.searchParams) || {};
+    const cookiesRead = await cookies();
```

Comment:
@doug-s-nava used `cookiesRead` and `*Read` a couple of times. Not wedded to that.